src/components/vm/consoles/serial.tsx

```diff
@@ -128,15 +128,18 @@ export const SerialMissing = ({
 }) => {
     const [inProgress, setInProgress] = useState(false);
 
-    function add_serial() {
+    async function add_serial() {
         setInProgress(true);
-        domainAttachSerialConsole(vm)
-                .catch(ex => onAddErrorNotification({
-                    text: cockpit.format(_("Failed to add serial console to VM $0"), vm.name),
-                    detail: ex.message,
-                    resourceId: vm.id,
-                }))
-                .finally(() => setInProgress(false));
+        try {
+            await virtXmlAdd(vm, "console", { type: "pty" });
+        } catch (ex) {
+            onAddErrorNotification({
+                text: cockpit.format(_("Failed to add serial console to VM $0"), vm.name),
+                detail: String(ex),
+                resourceId: vm.id,
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2334-da07d010-20250929-141718-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)